agent/xds/delta.go

```diff
@@ -434,9 +442,9 @@ func newDeltaType(
 // Recv handles new discovery requests from envoy.
 //
 // Returns true the first time a type receives a request.
-func (t *xDSDeltaType) Recv(req *envoy_discovery_v3.DeltaDiscoveryRequest, sf supportedProxyFeatures) bool {
+func (t *xDSDeltaType) Recv(req *envoy_discovery_v3.DeltaDiscoveryRequest, sf supportedProxyFeatures) (newSubscription, nack bool) {
```

Comment:
also nit: it would be cleaner for the second bool to be `ack`, so that `true, true` represents positive meanings "is new subscription, is ack-ed"

This would complicate the `if t == nil` check but I think that block can be removed (I don't see a codepath there this could be `nil` and it would be programming error anyways)

Reply:
An alternate idea would be making a new enum type:
```
type DeltaRecvResponse int

const (
	DeltaRecvResponseNACK DeltaRecvResponse = iota // zero value is nack which seems appropriate
	DeltaRecvResponseACK
	DeltaRecvResponseNewSubscription
)
```
Then you only have to return a single thing, instead of two bools.

Reply:
Ended up going with the enum. I added `DeltaRecvUnknownType` since we can end up with `t == nil` if Envoy decides to issue a request for a type we don't have a handler for.